### PR TITLE
Replacing all QRegExp with QRegularExpression

### LINKED ISSUE #961: Replace QRegExp with QRegularExpression
http://doc.qt.io/qt-5/qregularexpression.html#notes-for-qregexp-users

### PR DESCRIPTION
**Pull request by darkmattercoder**
**Detailed description**

According to #961, all `QRegExp` should get replaced by `QRegularExpression`. I did all that changes in the code and refacrored it so that it *should* work like before. However, as I am no user of `Cutter` and do not know, how to use the software, I am unable to know out of the box, what actions I have to perform, to verify the code changes. 

If you give me a hint, what I have to do in the interfaces to see the behaviour, I will of course examine the behaviour myself

closes #961

### CODE REVIEW

**Comment by xarkes** [comment]
Hello, thanks for your contribution!
One of us will test it before merging :)
You can also try to open Cutter on any file e.g. `./Cutter /bin/ls -A2` and play around with the classes you modified like ColorPicker or whatever.

**Comment by pelijah** [comment]
I am not good in regexp so I didn't fix it myself.
1) Style formatting changes should be moved to a separate PR.
2) I think it is not good idea to call `QRegularExpression` ctor in a loop.

File: src/common/HexHighlighter.cpp [hunk 1]
```diff
@@ -27,51 +27,53 @@ HexHighlighter::HexHighlighter(QTextDocument *parent)
...
     singleLineCommentFormat.setForeground(Qt::darkGreen);
-    rule.pattern = QRegExp(";[^\n]*");
+    rule.pattern = ";[^\n]*";
     rule.format = singleLineCommentFormat;
     highlightingRules.append(rule);
 
-    commentStartExpression = QRegExp("/\\*");
-    commentEndExpression = QRegExp("\\*/");
+    commentStartRegularExpression = "/\\*";
+    commentEndRegularExpression = "\\*/";
```
Comment (pelijah) [inline src/common/HexHighlighter.cpp]: This can affect performance too

**Comment by darkmattercoder** [comment]
I see the concerns regarding the style changes, and I did not notice, that my qt creator did that. I will change that again. A `clang.format` file for that in the repo would be a nice idea.

File: src/common/HexHighlighter.cpp (hunk 1 above)
Comment (darkmattercoder) [inline src/common/HexHighlighter.cpp]:
```suggestion
    commentStartRegularExpression = QStringLiteral("/\\*");
    commentEndRegularExpression = QStringLiteral("\\*/");
```
Do you have this in mind?

Comment (darkmattercoder) [reply src/common/HexHighlighter.cpp]:
```suggestion
    commentStartRegularExpression = QStringLiteral("/\\*");
    commentEndRegularExpression = QStringLiteral("\\*/");
```
Do you have this in mind?

**Comment by darkmattercoder** [comment]
It looks like someone else had a look into that once: https://forum.qt.io/topic/75232/qregexp-for-string-in-quotes/14

Probably I did a mistake and using `QRegularExpression` in the struct was caused by not explicitly including the header..... will re-check

File: src/common/HexHighlighter.cpp (hunk 1 above)
Comment (darkmattercoder) [reply src/common/HexHighlighter.cpp]: well, with 8979dddf1c0b1c3fd770ad2c272f4fced26654a8 that is obsolete

**Comment by karliss** [comment]
Why did Qt had to remove exeactMatch! The current solution is looks so ugly and unreadable, but I guess there isn't much we can do if their documentation says so.

**Pull request merged**